### ISSUE TITLE
How to Install Revit in Azure DevOps Pipeline

### ISSUE THREAD

**Issue by Dre-Tas** [DynamoDS/RevitTestFramework#105]
Hi everyone!

I'm getting started with CI with Azure DevOps (previously VSTS) and I'm using RTF for the testing of Revit-specific methods (which btw is really cool and I thank you a lot for it!)
I'm following the instructions [here](https://github.com/DynamoDS/RevitTestFramework/blob/Revit2019/docs/using_with_ci.md)

I now got my Pipeline up and running, connected to BitBucket and every time I push code to my repo the build starts and I run non-Revit tests automatically. All good until here.

But then I tried to implement the Revit testing bit and it gets confusing:
I guess the gist is somewhat clear, but I'm not too sure on how to run the Console app of RTF as it needs a version of Revit installed and I have no idea on how to install it on Azure!
I checked the CI instructions in the link above and it is very generic, mentioning the installation of Revit, but not much more.

Has anyone done this before? Or any advice on how to do this?

Cheers!

**Comment by AmeerMansourBeca** [comment]
hi 
did you manage to figure out a solution for this ?

**Comment by dosymep** [comment]
Hi, you can create own VM and setup Revit, and setup [self hosted BuildAgent](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser) on it.

**Comment by AmeerMansourBeca** [comment]
> Hi, you can create own VM and setup Revit, and setup [self hosted BuildAgent](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser) on it.

yes but how to make Revit run and gets the report back to Azure report or bitbucket ?

**Comment by Dre-Tas** [comment]
@AmeerMansourBeca  I'm about to start writing a paper + presentation for Autodesk University which is going to be at the end of September. I'll post it here when I have it ;)
(If I forget by the end of the year, please remind me)

**Comment by AmeerMansourBeca** [comment]
i think i am the one who will forget about it by that time hhhhh 😄 
can you give me a clue? i have VM and I was going to use azure self-agents
i was thinking of creating a batch file on the host that opens revit and sets RevitTestFramework to auto and then uploading the log file somewhere but I feel it is too much and I think there is an easier alternative.

**Comment by dosymep** [comment]
@AmeerMansourBeca, I think console application writes log in console, but build agent read console stream output

**Comment by dosymep** [comment]
@AmeerMansourBeca, I think so

**Comment by AmeerMansourBeca** [comment]
@dosymep 
i will try that and i will let you know 😄 
thanks

**Comment by dosymep** [comment]
@AmeerMansourBeca, maybe you should use this [method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.testcontext.writeline?view=visualstudiosdk-2022)
